src/chess/board.cc

```diff
@@ -982,8 +983,33 @@ void ChessBoard::SetFromFen(const std::string& fen, int* rule50_ply,
   string en_passant;
   int rule50_halfmoves;
   int total_moves;
-  fen_str >> board >> who_to_move >> castlings >> en_passant >>
-      rule50_halfmoves >> total_moves;
+  fen_str >> board;
+  if (fen_str.eof()) {
+    who_to_move = "w";
+  } else {
+    fen_str >> who_to_move;
+  }
+  if (fen_str.eof()) {
+    castlings = "KQkq";
+    ignore_castling_errors = true;
```

Comment:
I'm not sure about this part - the castling validation logic isn't strong enough to correctly filter castling fen component down to actual plausible scenarios.

Reply:
Should we assume "-" castling instead? The current implementation does have some oddness with differently positioned rooks that lichess/sf happily allows both sides to castle 😉 https://lichess.org/analysis/r2k3r/8/8/8/8/8/8/1R2K1R1

Reply:
assuming '-' seems safer if we're going to do this.

Our implementation only supports chess 960 castling - not castling with independently shuffled back ranks.

Reply:
Defaulting to - instead of KQkq needing ignores.